src/MyArrayList.java

```diff
@@ -33,21 +33,24 @@ public void addToArray(int number){
         }
     }
 
-    public void swapIndices (int index1, int index2){
-        int temp = this.numberArray[index2];
-        this.numberArray[index2] = this.numberArray[index1];
-        this.numberArray[index1] = temp;
-    }
-
-
 //    add at any index outside the scope. how many spaces will need to be passed?
     public void addToArrayAtIndex(int index, int number){
+        if(index > this.numberArray.length) {
+            for(int i = this.numberArray.length; i <= index; i++) {
+                expandArray(1);
```

Comment:
This is interesting, and I like that you used the For loop here, because you're showing understanding of how the For loop works.
Can you think of any way we could do this without a For loop? 
I can come discuss it with you later.